lib/std/unicode.zig

```diff
@@ -629,3 +629,71 @@ test "utf8ToUtf16LeWithNull" {
         testing.expect(utf16[2] == 0);
     }
 }
+
+pub fn utf8ToUtf16LeStringLiteral(comptime utf8: []const u8) *const [calcUtf16LeLen(utf8) :0] u16 {
+    comptime {
+        const len: usize = calcUtf16LeLen(utf8);
+        var utf16le: [len :0]u16 = [_ :0]u16{0} ** len;
+        const utf16le_len = utf8ToUtf16Le(&utf16le, utf8[0..]) catch |err| @compileError(err);
+        assert(len == utf16le_len);
+        return &utf16le;
+    }
+}
+
+fn calcUtf16LeLen(comptime utf8: []const u8) usize {
+    comptime {
+        var src_i: usize = 0;
+        var dest_len: usize = 0;
+        while (src_i < utf8.len) {
+            const n = utf8ByteSequenceLength(utf8[src_i]) catch @compileError(error.InvalidUtf8);
```

Comment:
I think this should be `catch unreachable`, and document in the function's doc comments that the function asserts valid UTF-8.

Reply:
I can make that change. The calculate function is `utf8ToUtf16Le` without the conversion, which was returning those errors in response to invalid data.